### ISSUE TITLE
Please fix warnings when -Weffc++

### ISSUE THREAD

**Issue by sheerluck** [jarro2783/cxxopts#216]
I set -Weffc++ in CMAKE_CXX_FLAGS_RELEASE and got bunch of warnings in my face :(

**Comment by dreuter** [comment]
I have a similar problem compiling `cxxopts` with `-Wconversion` on gcc 9. However I would like to propose instead of (or if desired in addition to) fixing these warnings (which may even depend on taste) to give the user the possibility to specify that they would like to include `cxxopts` as a system header, when adding it using `add_subdirectory`. This would suppress all warnings that are emitted by `cxxopts`.

If this would be an option for you I would gladly create a pull request for this.

Thanks for this great library btw :)

**Comment by cwreynolds** [comment]
Sorry if this is off-topic but I recently ran across some handy `#pragma` directives that can be wrapped around an `#include` to suppress warnings, by type. See an example here, avoiding compiler warnings when including headers for OpenCV: https://stackoverflow.com/a/59525150/1991373 (Note that `#pragma` is not part of the c++ standard.)

**Comment by dreuter** [comment]
Yes, I have also used that `#pragma` and it is especially useful because it also suppresses warnings that were generated by an instantiation of a template from the included header in your source code (which including as a system header does not).

However I currently compile on MSVC, gcc and clang and then this method of suppressing errors becomes quite verbose.
This could still be another possibility to add a `#ifdef CXXOPTS_DISABLE_ALL_WARNINGS` (which would tell cxxopts to use those `#pragma`s) to optionally suppress all warnings in cxxopts.
This [article](https://www.fluentcpp.com/2019/08/30/how-to-disable-a-warning-in-cpp/) has some information on how to do the `#pragma <compiler> diagnostic ignored ...` in a cross platform way.
